src/components/application_manager/src/application_state.cc

```diff
@@ -201,17 +207,24 @@ void ApplicationState::RemoveHMIState(const WindowID window_id,
     return;
   }
 
-  // unable to remove regular state
-  DCHECK_OR_RETURN_VOID(hmi_states.begin() != it);
-  HmiStates::iterator next = it;
-  HmiStates::iterator prev = it;
-  next++;
-  prev--;
-
-  if (next != hmi_states.end()) {
-    HmiStatePtr next_state = *next;
-    HmiStatePtr prev_state = *prev;
-    next_state->set_parent(prev_state);
+  EraseHMIState(hmi_states, it);
+}
+
+void ApplicationState::EraseHMIState(HmiStates& hmi_states,
+                                     HmiStates::iterator it) {
+  if (it == hmi_states.begin()) {
```

Comment:
```suggestion
  if (hmi_states.begin() == it) {
```

I think this follows the code style we have been using lately